src/views/EditPerson/SoldierScreen/SoldierForm/index.jsx

```diff
@@ -0,0 +1,415 @@
+            func: PersonValidate.zipCode,
+            message: "מיקוד צריך להכיל 7 ספרות",
+          },
+          {
+            func: PersonValidate.maxSeven,
+            message: "מיקוד צריך להכיל 7 ספרות",
+            avoidWrite: true,
+          },
+        ],
+      },
+      homeNumber: {
+        value: "",
+        errorMessage: "",
+        isValid: true,
+        validations: [
+          {
+            func: PersonValidate.homeNumber,
+            message: "",
+            avoidWrite: true,
+          },
+        ],
+      },
+      withClearance: {
+        value: "No classification",
+      },
+      job: {
+        value: "",
+      },
+      description: {
+        value: "",
+      },
+    });
+    initialRef.current = "Done";
+  }
+
+  // Each time a soldier reloads, their data is updated
+  useMemo(() => {
+    updateInputs({
+      firstName: {
+        value: soldier.firstName || "",
+        errorMessage: "",
+        isValid: true,
+      },
+      lastName: {
+        value: soldier.lastName || "",
+        errorMessage: "",
+        isValid: true,
+      },
+      identityCard: {
+        value: soldier.identityCard || "",
+        errorMessage: "",
+        isValid: true,
+      },
+      personalNumber: {
+        value: soldier.personalNumber || "",
+      },
+      clearance: {
+        value: soldier.clearance || "",
+      },
+      rank: {
+        value: soldier.rank || "",
+        errorMessage: "",
+        isValid: true,
+      },
+      phone: {
+        value: soldier.phone ? soldier.phone[0] : "",
+        errorMessage: "",
+        isValid: true,
+      },
+      mobilePhone: {
+        value: soldier.mobilePhone ? soldier.mobilePhone[0] : "",
+        errorMessage: "",
+        isValid: true,
+      },
+      mail: {
+        value: soldier.mail || "",
+        errorMessage: "",
+        isValid: true,
+      },
+      address: {
+        value: soldier.address || "",
+        errorMessage: "",
+        isValid: true,
+      },
+      withClearance: {
+        value: !soldier.clearance
+          ? "No classification"
+          : soldier.clearance === "0"
+          ? "No classification"
+          : "There is a classification",
+        errorMessage: "",
+        isValid: true,
+      },
+      job: {
+        value: soldier.job || "",
+        errorMessage: "",
+        isValid: true,
+      },
+      zipCode: {
+        value: "",
+        errorMessage: "",
+        isValid: true,
+      },
+      city: {
+        value: "",
+        errorMessage: "",
+        isValid: true,
+      },
+      description: {
+        value: "",
+        errorMessage: "",
+        isValid: true,
+      },
+      homeNumber: {
+        value: "",
+        errorMessage: "",
+        isValid: true,
+      },
+    });
+  }, [soldier, updateInputs]);
+
+  // Update Soldier to Kartoffel
+  const updateSoldier = (e) => {
+    // Get key value each field from useFormHandled
+    const fullPersonChange = handleSubmit(e);
```

Comment:
- rename `handleSubmit` to something else
- why use the event there? would it be simpler to take the `event.preventDefault();`  out of the handleSubmit function?